spec/services/messages/mention_service_spec.rb

```diff
@@ -5,69 +5,505 @@
   let!(:user) { create(:user, account: account) }
   let!(:first_agent) { create(:user, account: account) }
   let!(:second_agent) { create(:user, account: account) }
+  let!(:third_agent) { create(:user, account: account) }
+  let!(:admin_user) { create(:user, account: account, role: :administrator) }
   let!(:inbox) { create(:inbox, account: account) }
   let!(:conversation) { create(:conversation, account: account, inbox: inbox, assignee: user) }
+  let!(:team) { create(:team, account: account, name: 'Support Team') }
+  let!(:empty_team) { create(:team, account: account, name: 'Empty Team') }
   let(:builder) { double }
 
   before do
     create(:inbox_member, user: first_agent, inbox: inbox)
     create(:inbox_member, user: second_agent, inbox: inbox)
+    create(:team_member, user: first_agent, team: team)
+    create(:team_member, user: second_agent, team: team)
     conversation.reload
     allow(NotificationBuilder).to receive(:new).and_return(builder)
     allow(builder).to receive(:perform)
+    allow(Conversations::UserMentionJob).to receive(:perform_later)
   end
 
-  context 'when message contains mention' do
-    it 'creates notifications for inbox member who was mentioned' do
+  def expect_notification_for_user(user, message)
```

Comment:
lets not do method defintions here as its not a pattern which we generally use . lets call it directlly in the spec.